src/LanguageServer/IdePurescript/FileTypes.purs

```diff
@@ -0,0 +1,50 @@
+module LanguageServer.IdePurescript.FileTypes where
+
+import Prelude
+
+import Control.Alternative ((<|>))
+import Data.Map as Map
+import Data.Maybe (Maybe(..))
+import Data.String as String
+
+import LanguageServer.IdePurescript.Types (ServerState(..), ServerStateRec)
+import LanguageServer.Protocol.TextDocument (TextDocument, getText, getUri, getVersion)
+import LanguageServer.Protocol.Types (DocumentStore, DocumentUri(..))
+
+-- | Type representing files passsed by the IDE
+data RelevantFileType
+  = PureScriptFile
+  | JavaScriptFile
+  | UnsupportedFile
+
+derive instance Eq RelevantFileType
+instance Show RelevantFileType where
+  show PureScriptFile = "PureScriptFile"
+  show JavaScriptFile = "JavaScriptFile"
+  show UnsupportedFile = "UnsupportedFile"
+
+uriToRelevantFileType uri = f
+  where
+    extIs = uriExtensionIs uri
+    f
+      | extIs "purs" = PureScriptFile
+      | extIs "js" = JavaScriptFile
+      | otherwise = UnsupportedFile
+
+uriExtensionIs (DocumentUri uri) ext = ext' == after
+  where
+    ext' = "." <> ext
+    {after} = String.splitAt (String.length uri - String.length ext') uri
+
+jsUriToMayPsUri (DocumentUri str) =
+  (String.stripSuffix (String.Pattern ".js") str)
+  <#> (_ <> ".purs")
+  <#> DocumentUri
+
+
+tryGetDocument :: ServerState -> DocumentUri -> Maybe TextDocument
+tryGetDocument (ServerState ssr) uri =
```

Comment:
I don't understand why you would look up any of these, what's the point of this?

Reply:
In my corresponding PR to the VSCode plugin I've it to send changes made to JS files to the language server, but when those changes are received, they contain the text of the JS file; what I actually need is the contents of the corresponding PS, see here:

https://github.com/nwolverson/purescript-language-server/pull/181/files#diff-71c59da0078bac324d31359a59d13f21a7bfed2ce792c2ffee1ecb5035032079R306-R318

The cursory reading I've done on language servers suggests I should be relying on data coming from the client (IDE or whatever) not directly accessing the file system to get source code, so I'm using the existing state that should (and in my testing seems to) already contain the needed PS code. Based on the names of the fields my assumption is to first try looking looking in the fast rebuild queue, as it could sometimes be the case that if both the JS and corresponding PS file were changed; the presence of a queue implies changes are processed asynchronously, therefore, the data in parsedModules may be stale. Same reasoning for the diagnostics queue. Since I didn't come across documentation explicitly stating what those queues are for, I put a TODO comment to ask if this is correct.

Reply:
OK, I follow you, I think the way to do this is actually simpler and coming from none of those sources, the lsp libraries have a document store that we can access, we should use this:

https://github.com/nwolverson/purescript-language-server/blob/master/src/LanguageServer/Protocol/DocumentStore.purs#L12

That should represent the current document state, any `TextDocument` found in the other state is either for convenience or to capture the document state at the time of request.